web/components/auth-context.tsx

```diff
@@ -28,11 +28,11 @@ const ensureDeviceToken = () => {
   return deviceToken
 }
 
-export const AuthContext = createContext<AuthUser>(null)
-
-export function AuthProvider({ children }: any) {
-  const [authUser, setAuthUser] = useStateCheckEquality<AuthUser>(undefined)
+export const AuthContext = createContext<AuthUser>(undefined)
 
+export function AuthProvider(props: { children: ReactNode; user?: AuthUser }) {
+  const { children, user } = props
+  const [authUser, setAuthUser] = useStateCheckEquality<AuthUser>(user)
   useEffect(() => {
     const cachedUser = localStorage.getItem(CACHED_USER_KEY)
     setAuthUser(cachedUser && JSON.parse(cachedUser))
```

Comment:
If `user` was passed as a prop, shouldn't we prefer it to the one in localStorage?

Reply:
Yeah, I guess so? It seems like this shouldn't be relevant to the normal case, but if the two disagree, the server one is certainly more up to date, so there's no reason to use the local storage one.

I should think about what to actually do with this local storage user cache.